src/Conversions/Manipulations.php

```diff
@@ -52,6 +60,34 @@ public function isEmpty(): bool
     public function apply(ImageDriver $image): void
     {
         foreach ($this->manipulations as $manipulationName => $parameters) {
+            switch($manipulationName){
```

Comment:
Could you use a `match` operator instead?

Reply:
What do you think about using neither match nor switch. Just recreate all Enums if the parameter is present independent of thy type of manipulation
```suggestion
    (isset($parameters['type'])) && $parameters['type'] = BorderType::from($parameters['type']) ;
    (isset($parameters['fit'])) && $parameters['fit'] = Fit::from($parameters['fit']);
    (isset($parameters['position'])) && $parameters['position'] = AlignPosition::from($parameters['position']);
    (isset($parameters['colorFormat'])) && $parameters['colorFormat'] = ColorFormat::from($parameters['colorFormat']);
    (isset($parameters['constraints'])) && $parameters['constraints'] = Constraint::from($parameters['constraints']);
    (isset($parameters['position'])) && $parameters['position'] = CropPosition::from($parameters['position']);
    (isset($parameters['fit'])) && $parameters['fit'] = Fit::from($parameters['fit']);
    (isset($parameters['flip'])) && $parameters['flip'] = FlipDirection::from($parameters['flip']);
```

Reply:
I think a match we be more clear in this case